pydantic/utils.py

```diff
@@ -413,18 +413,26 @@ class GetterDict(Representation):
     We can't inherit from Mapping[str, Any] because it upsets cython so we have to implement all methods ourselves.
     """
 
-    __slots__ = ('_obj',)
+    __slots__ = ('_obj', '_kwargs')
 
-    def __init__(self, obj: Any):
+    def __init__(self, obj: Any, kwargs: Optional[Dict[str, Any]] = None):
         self._obj = obj
+        self._kwargs = kwargs or None
 
     def __getitem__(self, key: str) -> Any:
         try:
+            if self._kwargs and key in self._kwargs:
+                return self._kwargs[key]
```

Comment:
IMO, this should be out of `try/except AttrubuteError` block.

And perhaps we should ask for forgiveness, not permission (like you've done it in `get` method)?